File: client/src/types/JsonModels/WorkoutTemplates/WorkoutTemplateExerciseModel.ts
import type { JsonModels } from "../../backend";
import type { ExerciseLoadBasis } from "../Enums/ExerciseLoadBasis";

export interface WorkoutTemplateExerciseModel
{
	id: number;
	exerciseId: number;
	exerciseName: string;
	exerciseImageUrl?: string;
	exerciseLoadBasis?: ExerciseLoadBasis;
	orderIndex: number;
	targetSets: number;
	targetReps?: string;
	targetWeightKg?: number;
	targetRestSeconds?: number;
	tempo?: string;
	notes?: string;
	sets: JsonModels.WorkoutTemplates.WorkoutTemplateExerciseSetModel[];
}

File: server/FitMate.Core/JsonModels/WorkoutTemplates/WorkoutTemplateExerciseModel.cs
namespace FitMate.Core.JsonModels.WorkoutTemplates;

public class WorkoutTemplateExerciseModel
{
    public long Id { get; set; }
    public long ExerciseId { get; set; }
    public string ExerciseName { get; set; } = string.Empty;
    public string? ExerciseImageUrl { get; set; }
    public int OrderIndex { get; set; }
    public int TargetSets { get; set; }
    public string? TargetReps { get; set; }
    public decimal? TargetWeightKg { get; set; }
    public int? TargetRestSeconds { get; set; }
    public string? Tempo { get; set; }
    public string? Notes { get; set; }
    public List<WorkoutTemplateExerciseSetModel> Sets { get; set; } = [];
}
